pacsman/filesystem_dev_client.py

```diff
@@ -144,10 +147,19 @@ def date_filter(study_ds):
             else:
                 study_date_str = None
 
-            if study_start_date is None or study_end_date is None or study_date_str is None:
+            if (study_start_date is None and study_end_date is None) or study_date_str is None:
                 return True
+
             study_date = datetime.strptime(study_date_str, date_format_str).date()
-            return study_date >= study_start_date and study_date <= study_end_date
+            if study_start_date:
+                if study_end_date:
+                    return study_date >= study_start_date and study_date <= study_end_date
+                else:
+                    return study_date >= study_start_date
+
+            # Should already be covered by edge-case early return
+            assert study_end_date is not None
+            return study_date <= study_end_date
```

Comment:
This is just a preference of mine, but I like the visual symmetry of keeping all these cases at max 2 levels of indention. e.g. 

```py
if study_start date and study_end date:
    ...
elif study_start_date: 
    ...
else:
    assert study_end_date is not None
    ...
```

You only are adding one additional line by structuring this way, and I think it may be slightly easier to parse if you're reading the function for the first time. I think it also makes it clearer that only one of these code paths will be executed (although that is already pretty clear in this case since they're all return statements)

Reply:
Good idea; I concur that your flow is more readable. Updated in 559813f67229eb6d94c5ca9166ca844339a97f55.